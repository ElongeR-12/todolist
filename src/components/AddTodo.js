import React, {useState} from 'react';
const AddTodo = ({todos, setTodos}) => {
    const [title, setTitle] = useState("")
    const [description, setDescription] = useState("")
    const addTodo = async (todo) => {
        const res = await fetch('http://localhost:5000/todos', {
          method: 'POST',
          headers: {
            'Content-type': 'application/json',
          },
          body: JSON.stringify(todo),
        })
        const data = await res.json()
        setTodos([data, ...todos])
      }
    const handleSubmit =(e)=>{
        e.preventDefault();
        title && addTodo({description: description ? description:"", title: title, state: false});
        setTitle("");
        setDescription("")
    }
  return (
    <form onSubmit={handleSubmit}>
        <input
            type="text"
            value={title}
            placeholder="Enter a title..."
            onChange={e => setTitle(e.target.value)}
            required
        />
        <input
            type="text"
            value={description}
            placeholder="Enter a description..."
            onChange={e => setDescription(e.target.value)}
        />
        <button type="submit">add todo</button>
    </form>
  )
}

export default AddTodo