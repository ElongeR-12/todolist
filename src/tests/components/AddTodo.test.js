import React from 'react';
import { shallow } from 'enzyme';
import AddTodo from '../../components/AddTodo';
describe("AddTodo test",()=>{
    test('should test AddTodo component', () => {
        const wrapper = shallow(<AddTodo/>);
        expect(wrapper).toMatchSnapshot();
      });
})