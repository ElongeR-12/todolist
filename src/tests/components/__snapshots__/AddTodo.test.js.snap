// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTodo test should test AddTodo component 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Enter a title..."
    required={true}
    type="text"
    value=""
  />
  <input
    onChange={[Function]}
    placeholder="Enter a description..."
    type="text"
    value=""
  />
  <button
    type="submit"
  >
    add todo
  </button>
</form>
`;
